pywinauto/linux/atspi_element_info.py

```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+# GUI Application automation and testing library
+# Copyright (C) 2006-2019 Mark Mc Mahon and Contributors
+# https://github.com/pywinauto/pywinauto/graphs/contributors
+# http://pywinauto.readthedocs.io/en/latest/credits.html
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of pywinauto nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Linux AtspiElementInfo class"""
+
+from .atspi_objects import AtspiAccessible, AtspiComponent, AtspiStateEnum, AtspiAction, AtspiValue, \
+    IATSPI
+from ..element_info import ElementInfo
+
+
+class AtspiElementInfo(ElementInfo):
+
+    """Wrapper for window handler"""
+
+    atspi_accessible = AtspiAccessible()
+
+    def __init__(self, handle=None):
+        """Create element by handle (default is root element)"""
+        if handle is None:
+            self._handle = self.atspi_accessible.get_desktop(0)
+        else:
+            self._handle = handle
+
+    def __get_elements(self, root, tree, **kwargs):
+        tree.append(root)
+        for el in root.children(**kwargs):
+            self.__get_elements(el, tree, **kwargs)
+
+    def __hash__(self):
+        """Return a unique hash value based on the element's handle"""
+        return hash(self.handle)
+
+    def __eq__(self, other):
+        """Check if two AtspiElementInfo objects describe the same element"""
+        if self.control_type == "Application":
+            return self.process_id == other.process_id
+        return self.rectangle == other.rectangle
```

Comment:
Maybe add `and self.handle == other.handle` at the end.

Reply:
Hmm... It breaks the following test: 
```Python
            desktop = AtspiElementInfo()
            desktop1 = AtspiElementInfo()
            assert desktop = desktop1
```
That's because the constructor creates two different AtspiAccessible objects under the hood.